cpp/log/cert.cc

```diff
@@ -380,6 +475,26 @@ StatusOr<bool> Cert::IsSignedBy(const Cert& issuer) const {
     ClearOpenSSLErrors();
     return false;
   }
+
+  // With BoringSSL we might have a signature algorithm that is not supported
+  // by X509_verify but we still want to accept into a log. This is a weaker
+  // check than x509_verify but sufficient for our needs as we are rejecting
+  // spam rather than intending to trust the certificate.
+  if (lib == ERR_LIB_EVP && reason == EVP_R_UNSUPPORTED_ALGORITHM) {
```

Comment:
I'm not sure how much stability BoringSSL promises wrt error codes but it's generally a pain that applications rely on them and it might be more robust to just always attempt the DSA verify; it'll bail quickly if the key type is wrong.